rita_common/src/peer_listener/mod.rs

```diff
@@ -103,36 +145,38 @@ fn listen_to_available_ifaces(peer_listener: &mut PeerListener) {
                         .interfaces
                         .insert(new_listen_interface.ifname.clone(), new_listen_interface);
                 }
-                Err(_e) => {}
+                Err(e) => {
+                    error!("Received an error while listening to interfaces: {:?}", e)
+                }
             }
         }
     }
 }
 
 /// Ticks the peer listener module sending ImHere messages and receiving Hello messages from all
 /// peers over UDP
-pub fn peerlistener_tick() {
+pub fn peerlistener_tick() -> PeerListener {
     trace!("Starting PeerListener tick!");
 
     let mut writer = PEER_LISTENER.write().unwrap();
```

Comment:
we're holding this lock while performing operations, if we end up blocking trying to recieve an I'm here it will be problematic, is there any way we can eliminate that? Maybe by getting a copy of the data we need and then integrating the response with getter/setter functions

Reply:
updated with getter/setters